test/widgets/message_list_test.dart

```diff
@@ -51,13 +52,18 @@ void main() {
     List<User>? users,
     List<Subscription>? subscriptions,
     UnreadMessagesSnapshot? unreadMsgs,
+    int? zulipFeatureLevel,
+    bool skipPumpAndSettle = false,
   }) async {
     addTearDown(testBinding.reset);
     streams ??= subscriptions ??= [eg.subscription(eg.stream(streamId: eg.defaultStreamMessageStreamId))];
     await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
       streams: streams, subscriptions: subscriptions, unreadMsgs: unreadMsgs));
     store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
     connection = store.connection as FakeApiConnection;
+    if (zulipFeatureLevel != null) {
+      connection.zulipFeatureLevel = zulipFeatureLevel;
```

Comment:
Hmm, I think `eg.selfAccount` carries a different `zulipFeatureLevel` if we only override the feature level on the `connection`.